lib/src/facts/posix/identity_resolver.cc

```diff
@@ -25,7 +25,15 @@ namespace facter { namespace facts { namespace posix {
         uid_t uid = geteuid();
         struct passwd pwd;
         struct passwd *pwd_ptr;
-        int err = getpwuid_r(uid, &pwd, buffer.data(), buffer.size(), &pwd_ptr);
+        int err = -1;
+
+        do {
+            err = getpwuid_r(uid, &pwd, buffer.data(), buffer.size(), &pwd_ptr);
+            if (err == ERANGE) {
+                buffer.resize(buffer.size() + 1024);
```

Comment:
*Typically* I'd prefer doubling the buffer space each time, since that will approach the needed space in logarithmic time (at the cost of using extra memory when it inevitably overshoots).

In this case, I'm fine with the linear growth. The difference in complexity only affects pwd entries that are greater than 3KiB - and probably not measurably until 7KiB, where this version (theoretically) takes twice as long as a doubling one would.

Since even entries that are greater than 1KiB appear to be pretty rare, I'm fine with the linear algorithm here. No sense creating busy work for you over an optimization that won't matter in the real world.